src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,147 @@
+package duke;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+
+
+/**
+ * Provides a class to interprets user inputs and converts them into Commands
+ * to be executed.
+ */
+public class Parser {
+
+    /**
+     * Represents date and time.
+     */
+    private static final SimpleDateFormat DATE_TIME_FORMAT = new SimpleDateFormat("dd/MM/yyyy HHmm");
+    /**
+     * Represents date.
+     */
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy");
+
+    /**
+     * Returns different Commands depending on the keywords that the user inputs.
+     * These keywords include "bye", "list", "done", "delete", "todo",
+     * "event", "deadline". Words that are not recognised will not be read
+     * and the function will ask the user to input a valid keyword.
+     *
+     * @param userInput Input that the user typed in.
+     * @return Commands depending on which keyword the user used.
+     * @throws DukeException If user entered an invalid input.
+     */
+    public Command parse(String userInput) throws DukeException {
+        String[] userInputArr = userInput.split(" ");
+        String userKeyword = userInputArr[0];
+        try {
+            switch (userKeyword) {
+            case "bye":
+                return new ExitCommand();
+
+            case "list":
+                return new ListCommand();
+
+            case "find":
+                if (userInput.length() < 2) {
+                    throw new DukeException("Please enter a keyword to find\n");
+                }
+                String keyword = userInput.substring(5);
+                return new FindCommand(keyword);
+
+            case "done":
+            case "delete":
+                if (userInputArr.length < 2) {
+                    throw new DukeException("Please enter a task number to be selected\n");
+                }
+                String taskNumber = userInputArr[1];
+
+                int taskNumberInt = Integer.parseInt(taskNumber) - 1;
+                if (userKeyword.equals("done")) {
+                    return new DoneCommand(taskNumberInt);
+                } else {
+                    return new DeleteCommand(taskNumberInt);
+                }
+
```

Comment:
Perhaps this line break can be omitted? I noticed double line breaks in several other places too.